smart-contracts/contracts/mixins/MixinEventHooks.sol

```diff
@@ -10,44 +12,72 @@ import './MixinLockCore.sol';
 contract MixinEventHooks is
   MixinLockCore
 {
-  ILockEventHooks public onKeySoldHook;
-  ILockEventHooks public onKeyCancelHook;
+  using Address for address;
+
+  ILockKeyPurchaseHook public onKeyPurchaseHook;
+  ILockKeyCancelHook public onKeyCancelHook;
 
   /**
    * @notice Allows a lock manager to add or remove an event hook
    */
   function setEventHooks(
-    address _onKeySoldHook,
+    address _onKeyPurchaseHook,
     address _onKeyCancelHook
   ) external
     onlyLockManager()
   {
-    onKeySoldHook = ILockEventHooks(_onKeySoldHook);
-    onKeyCancelHook = ILockEventHooks(_onKeyCancelHook);
+    require(_onKeyPurchaseHook == address(0) || _onKeyPurchaseHook.isContract(), 'INVALID_ON_KEY_SOLD_HOOK');
```

Comment:
I can't remember from the meeting yesterday why `_onKeyPurchaseHook == address(0)` is a valid condition here?

Reply:
Doh. nevermind. This is to Unset an existing hook.